core/src/main/java/io/questdb/griffin/engine/table/CompiledFilterRecordCursor.java

```diff
@@ -189,26 +192,13 @@ public long size() {
         return -1;
     }
 
-    private boolean nextColTopsRow() {
-        seekNextColTopsRow();
-        if (current < hi) {
-            return true;
-        }
-        return nextPage();
-    }
-
-    private void seekNextColTopsRow() {
-        while (++current < hi) {
-            recordA.setIndex(current);
-            if (colTopsFilter.getBool(recordA)) {
-                return;
-            }
-        }
+    private long getCurrentRowIndex() {
+        return hasDescendingOrder ? (hi - current - 1) : current;
```

Comment:
I cannot find a test which covers branch when `hasDescendingOrder == true`
I added `assert !hasDescendingOrder;`, ran all tests with `-ea` and all passed.

Reply:
I saw debugger only enter the hasDescendingOrder == true branch :) 
Turns out casting used in my generic test disabled jit so I removed it and saw that both branches are used (in latest commit) .